invisible_cities/reco/trigger_functions.py

```diff
@@ -0,0 +1,198 @@
+
+    # Signal above threshold, allowing for pulse_ext counts below threshold
+    triggers   = []
+    indexes    = np.where(signal > hrange[0])[0]
+    if len(indexes) == 0: return triggers # Protection in case min. threshold is not surpassed
+    index_sel  = np.diff(indexes)>(pulse_ext+1)
+    candidates = np.split(indexes, np.where(index_sel)[0] + 1)
+
+    for p in candidates:
+        peak_slice = slice(p[0]-1, p[-1] + pulse_ext + 1) # DAQ gets pulse_ext counts after peaks; also, for Q it takes a bin before TODO Fix for N100 (remove -1)
+        peak       = signal[peak_slice]
+        if not (len(peak) > max(discard_width, 1)): continue
+        width      = len(peak) - 1           # TODO Remove -1 in N100
+        height     = peak[1:  ].max()        # TODO Full peak for N100
+        peak       = peak[ :-2][peak[:-2]>0] # Only counts above 0 are used for Q, last bin not considered TODO ONLY -1 for N100
+        charge     = peak.sum()
+        if len(peak)==0: continue
+        start_time = p[0]
+
+        # Trigger validity
+        trigger_w   = in_range(width , *wrange, left_closed=True , right_closed=False)
+        trigger_q   = in_range(charge, *qrange, left_closed=False, right_closed=False)
+        trigger_h   = height < hrange[1]
+        trigger_all = all([trigger_w, trigger_q, trigger_h])
+
+        # Baseline at the start of the pulse
+        baseline_t0 = baseline[start_time]
+
+        trigger     = TriggerInfo(event_id   , pmt_id
+                                 ,start_time , charge        , width    , height
+                                 ,trigger_q  , trigger_w     , trigger_h,   True, trigger_all  # By default multiflag is set as true
+                                 ,baseline_t0, wvf_max_height
+                                 ,-1         , -1            , -1) # Filled later in the coincidence module
+        triggers  .append(trigger)
+
+    # Extra-peaks protection
+    if multipk is not None:
+        # Evaluate if later peaks (must be over discard_width) are candidates for the multipeak protection
+        for i, t0 in zip(range(len(triggers[:-1])), triggers[:-1]):
+            new_val = t0.valid_peak
+            for t1 in triggers[i+1:]:
+                multi_candidate = ((t1.width        >=                              multi_w  ) &
+                                   (t1.q            >                               multi_q  ) &
+                                   (t1.trigger_time <= t0.trigger_time + t0.width + multi_t+2)) # Start of trigger + peak width + time window + FPGA delay (2)
+                if not multi_candidate:
+                    continue
+                # In case the later peak ends after the multipeak protection window ends
+                if (t1.trigger_time + t1.width) > (t0.trigger_time + t0.width + multi_t + 2): # Event partially contained in the multipeak window
+                    peak_slice = slice(t1.trigger_time - 1, t0.trigger_time + t0.width + multi_t + 2 + 1)
```

Comment:
`t0.trigger_time + t0.width + multi_t + 2` seems to be a relevant value. It's worth to give it a name for readability.